src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.IGroupedAnalyzerActions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal abstract partial class AnalyzerDriver
+    {
+        protected abstract IGroupedAnalyzerActions EmptyGroupedActions { get; }
+        protected abstract IGroupedAnalyzerActions CreateGroupedActions(DiagnosticAnalyzer analyzer, in AnalyzerActions analyzerActions);
+
+        protected interface IGroupedAnalyzerActions
```

Comment:
Why do we need this interface?

Reply:
`AnalyzerDriver` is the language agnostic part of the implementation, that has language specific `AnalyzerDriver<TLanguageKindEnum>` implementations. `GroupedAnalyzerActions` is defined in the sub-type, but we need to invoke/cache these grouped actions in the base language agnostic `AnalyzerDriver` type. I plan to merge `AnalyzerDriver` into `AnalyzerDriver<TLanguageKindEnum>` in a future PR, that will allow us to remove this interface and cleanup code.